### ISSUE TITLE
Questions about Activation Tuning effectiveness and neuron-selective fine-tuning implementation

### ISSUE THREAD

**Issue by Nryu0630** [Leey21/InverTune#1]
Hi, thank you for releasing the code for InverTune.

I am trying to reproduce the BadNets defense results on CLIP RN50 following the attack setting in the paper. I trained a backdoored RN50 model with BadNets, using `mushroom` as the target class.

The original backdoored model seems to be successfully trained. Its evaluation results are approximately:

```text
ASR: 99.47%

zero-shot accuracy: 58.94%
```
Then I performed trigger inversion. When evaluating the original backdoored model with the inverted trigger, the inverted trigger can also activate the backdoor successfully. The results are approximately:
```text
target_idx: 947
target_class: n07734744
target_prompt: mushroom
clean_accuracy_percent: 58.942
trigger_target_rate_percent: 94.928
```
However, after running Activation Tuning with the paper setting, the ASR is still around 87%. The backdoor is reduced only slightly, instead of being largely mitigated.
My Activation Tuning configuration is:
```text
num_steps: 1
learning_rate: 0.01
save_interval: 100
ft_epochs: 800
candidate_layers:
  - model.visual.layer1
  - model.visual.layer2
  - model.visual.layer3
  - model.visual.layer4
```
The evaluation result after Activation Tuning is approximately:
```text
asr_percent: 87.70
backdoor_accuracy: 46.008
```
So I would like to ask:
1.    Could you please help me identify what might be wrong in my reproduction, and how I can reproduce the defense results reported in the paper?
2. In Algorithm 2, Activation Tuning is described as neuron-selective fine-tuning, where parameter masks and neuron-masked gradients are used to update only parameters corresponding to critical neurons. However, in the released ActivationTuning.py, the optimizer seems to be constructed over all model parameters:
```text
optimizer = torch.optim.AdamW(model.parameters(), lr=2e-6, weight_decay=1e-4)
```

<img width="730" height="99" alt="Image" src="https://github.com/user-attachments/assets/1dbd06df-afdd-460a-8722-db1af222ebd9" />

The critical neuron mask seems to be applied only to the activation alignment loss. I did not find gradient masking or parameter masking before optimizer.step().

Is the released implementation intended to perform full-model fine-tuning guided by masked activation loss, or is the gradient/parameter mask for critical neurons missing from the released code?
